### PR TITLE
feat: IDP approver roles for BCeID, GitHub and BC Services Card 

### PR DESCRIPTION
**Pull request by NithinKuruba**

### CODE REVIEW

**Review by thegentlemanphysicist — APPROVED** [review]

File: app/jest/idp-approvals.test.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+  it('GitHub Approver', async () => {
+    jest
+      .spyOn(requestModule, 'getRequestAll')
+      .mockImplementationOnce(() => Promise.resolve([{ count: 1, rows: [sampleRequests()[1]] }, null]));
+    jest.spyOn(requestModule, 'updateRequest').mockImplementation(() => Promise.resolve([{}, null]));
+    jest
+      .spyOn(eventModule, 'getEvents')
+      .mockImplementation(() => Promise.resolve([{ count: 1, rows: sampleEvents() as any }, null]));
+    render(<AdminDashboard session={sampleSession} onLoginClick={jest.fn} onLogoutClick={jest.fn} />);
```
Comment (jlangy) [inline app/jest/idp-approvals.test.tsx:219]: Is the sample session changing to have the github role for this test, or does it still have the bceid-approver role only? I didn't see where it updates

**Review by jlangy — COMMENTED** [review]
I left a comment there was one part of the unit test I didn't fully understand

File: app/jest/idp-approvals.test.tsx (hunk 1 above)
Comment (NithinKuruba) [reply app/jest/idp-approvals.test.tsx]: The sample session was just added to ensure users with approver roles are permitted to see admin dashboard as per new validation. The getRequestAll is mocked to return IDP specific requests but when deployed, the function uses approver roles to return IDP specific requests.

Comment (jlangy) [reply app/jest/idp-approvals.test.tsx]: Ah I see, sounds good!

**Review by jlangy — APPROVED** [review]

**Pull request merged**